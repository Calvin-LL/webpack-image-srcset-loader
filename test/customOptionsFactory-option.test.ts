import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe.each([4, 5] as const)(
  'v%d "customOptionsFactory" option',
  (webpackVersion) => {
    test("should work with a function", async () => {
      const mockCustomOptionsFactory = jest.fn().mockReturnValue({
        fileLoaderOptions: {
          esModule: false,
        },
      });
      const compiler = getCompiler(webpackVersion, {
        sizes: ["2x", null, "1x", "300w"],
        customOptionsFactory: (
          width: number | undefined,
          scale: number | undefined,
          existingOptions: object
        ) => mockCustomOptionsFactory(width, scale, existingOptions),
      });
      const stats = await compile(webpackVersion, compiler);

      expect(
        execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
      ).toMatchSnapshot("result");

      expect(mockCustomOptionsFactory).toHaveBeenCalled();
      expect(mockCustomOptionsFactory).toMatchSnapshot();
    });
  }
);
