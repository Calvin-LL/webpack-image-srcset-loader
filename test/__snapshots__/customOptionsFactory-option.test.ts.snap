// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 "customOptionsFactory" option should work with a function 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      undefined,
      1,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      0.5,
      undefined,
    ],
    Array [
      300,
      undefined,
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
  ],
}
`;

exports[`v4 "customOptionsFactory" option should work with a function: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2x, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 1x, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 300w"`;

exports[`v5 "customOptionsFactory" option should work with a function 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      undefined,
      1,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      0.5,
      undefined,
    ],
    Array [
      300,
      undefined,
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
  ],
}
`;

exports[`v5 "customOptionsFactory" option should work with a function: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2x, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 1x, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 300w"`;
